upgrade/helpers/openstack.py

```diff
@@ -36,98 +38,105 @@ def get_openstack_client():
-    # Validate ssh_key is added into openstack project
-    openstack_client.keypairs.find(name=ssh_key)
-    image = openstack_client.images.find(name=image_name)
-    flavor = openstack_client.flavors.find(name=flavor_name)
-    network = openstack_client.networks.find(label=network_name)
-    floating_ip = openstack_client.floating_ips.create(
-        openstack_client.floating_ip_pools.list()[0].name
-    )
+    env.user = 'root'
+    env.disable_known_hosts = True
+    if flavor_name is None:
+        flavor_name = os.environ.get('FLAVOR_NAME')
+    if network_name is None:
+        network_name = os.environ.get('NETWORK_NAME')
+    if ssh_key is None:
+        ssh_key = os.environ.get('OSP_SSHKEY')
+    openstack_client = shade.openstack_cloud(cloud='satellite-jenkins')
+    # Validate image is added into openstack project
+    image = openstack_client.get_image(image_name)
     # Create instance from the given parameters
     logger.info('Creating new Openstack instance {0}'.format(instance_name))
-    instance = openstack_client.servers.create(
+    instance = openstack_client.create_server(
         name=instance_name,
         image=image.id,
-        flavor=flavor.id,
+        flavor=flavor_name,
         key_name=ssh_key,
-        network=network.id
+        network=network_name,
+        volume_size=volume_size,
+        wait=True
     )
-    # Assigning floating ip to instance
-    timeup = time.time() + int(timeout) * 60
-    while True:
-        if time.time() > timeup:
-            logger.warning(
-                'The timeout for assigning the floating IP has reached!')
-            sys.exit(1)
-        try:
-            instance.add_floating_ip(floating_ip)
-            logger.info('SUCCESS!! The floating IP {0} has been assigned '
-                        'to instance!'.format(floating_ip.ip))
-            break
-        except novaclient.exceptions.BadRequest:
-            time.sleep(5)
+    if instance.interface_ip:
+        ip_addr = instance.interface_ip
+    else:
+        logger.error("No floating Ip assigned")
+
     # Wait till DNS resolves the IP
-    logger.info('Pinging the Host by IP:{0} ..........'.format(floating_ip.ip))
-    host_pings(str(floating_ip.ip))
+    logger.info('Pinging the Host by IP:{0} ..........'.format(ip_addr))
+    host_ssh_availability_check(ip_addr)
+    host_pings(str(ip_addr))
     logger.info('SUCCESS !! The given IP has been pinged!!\n')
     logger.info('Now, Getting the hostname from IP......\n')
-    hostname = get_hostname_from_ip(str(floating_ip.ip))
+    hostname = get_osp_hostname(ip_addr)
     if not hostname:
         sys.exit(1)
-    env['{0}_host'.format(product)] = hostname
     logger.info('Pinging the Hostname:{0} ..........'.format(hostname))
     host_pings(hostname)
     logger.info('SUCCESS !! The obtained hostname from IP is pinged !!')
     # Update the /etc/hosts file
+    execute(lambda: run("hostnamectl set-hostname {0}".format(
```

Comment:
You are setting the hostname on an instance, but before that, you are running ```execute``` on a hostname. Do you think that will work? Have you tested it ?

Reply:
Its bit confusing, Please explain more about it !

Reply:
I am setting the hostname, just to be in sync with DNS not actually required but good to have. :)
Yes, it is tested